geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/netty/Coder.java

```diff
@@ -344,6 +347,15 @@ public static long bytesToLong(byte[] bytes) {
    * @throws NumberFormatException if bytes to string does not yield a convertible double
    */
   public static double bytesToDouble(byte[] bytes) {
+    if (isPositiveInfinity(bytes)) {
+      return POSITIVE_INFINITY;
+    }
+    if (isNegativeInfinity(bytes)) {
+      return NEGATIVE_INFINITY;
+    }
+    if (isNaN(bytes)) {
+      throw new NumberFormatException(ERROR_NOT_A_VALID_FLOAT);
```

Comment:
Strictly speaking, I think this should return `Double.NaN` rather than throwing, as that is a valid value that a Double can take. It'll be the responsibility of the caller to determine what to do with the `NaN` value in that case.